<a href="https://colab.research.google.com/github/ElofssonLab/kb8029-book/blob/main/notebooks/day01-discussion-3.ipynb" style="display:inline-block;padding:10px 18px;background-color:#F9AB00;color:#000000;font-weight:bold;text-decoration:none;border-radius:6px;font-family:sans-serif;font-size:14px;">&#9654;&nbsp; Open in Google Colab</a>

# Day 1 — In-class discussion problem (3 of 3)

Work this out **by hand in your group first** — then run the code cell
to check your answer before presenting.

## Segment 3: How much should you trust a plagiarism-detection flag?

A university's automated system flags 85% of truly plagiarized
assignments ($P(\text{flag}\mid\text{plagiarized}) = 0.85$), and
incorrectly flags 8% of genuine assignments
($P(\text{flag}\mid\text{not plagiarized}) = 0.08$). Historically,
about 3% of submitted assignments are actually plagiarized.

**As a group:**

1. Write down what each number above means, and what you're being asked to find, in $P(\cdot\mid\cdot)$ notation. Which direction of conditioning does the student care about?
2. Count it out: of 10,000 assignments, how many are plagiarized, how many of those are flagged, and how many genuine assignments are flagged? What fraction of the flagged assignments is plagiarized, $P(\text{plagiarized}\mid\text{flag})$?
3. Would you recommend automatically failing every flagged student based on this number alone? Why or why not?

Run the cell below to check your answer to (2). (If you read the optional Bayes box on the Day 1 page: the same number comes out of Bayes' rule.)

In [1]:
n = 10_000
plagiarized = 0.03 * n
genuine = n - plagiarized
flagged_plag = 0.85 * plagiarized
flagged_genuine = 0.08 * genuine

print(f"plagiarized: {plagiarized:.0f}, flagged among them: {flagged_plag:.0f}")
print(f"genuine: {genuine:.0f}, flagged among them: {flagged_genuine:.0f}")
p_plag_given_flag = flagged_plag / (flagged_plag + flagged_genuine)
print(f"P(plagiarized | flag) = {flagged_plag:.0f} / {flagged_plag + flagged_genuine:.0f} = {p_plag_given_flag:.3f} = {p_plag_given_flag*100:.1f}%")

plagiarized: 300, flagged among them: 255
genuine: 9700, flagged among them: 776
P(plagiarized | flag) = 255 / 1031 = 0.247 = 24.7%


**Discussion point:** despite a detector that catches 85% of cases, a
flagged assignment is plagiarized only about 25% of the time. Plagiarism
is rare, so the false positives from the large honest majority (776)
outnumber the true positives from the small dishonest minority (255).
$P(\text{flag}\mid\text{plagiarized})$ and $P(\text{plagiarized}\mid\text{flag})$
are very different numbers: the direction of conditioning matters.